capstone/capapi/resources.py

```diff
@@ -1,50 +1,45 @@
-import os
 from datetime import datetime
 import logging
+import zipfile
+import tempfile
 
 from django.conf import settings
 from django.core.mail import send_mail
 from django.template.defaultfilters import slugify
 
-logger = logging.getLogger(__name__)
-
+from capdb.models import CaseXML
 
-def get_formatted_date():
-    return slugify(str(datetime.today()))
+logger = logging.getLogger(__name__)
 
 
-def format_filename(case_id, whitelisted=False):
-    cdir, cpgnumber = case_id.split('_')
-    cdirname = cdir + '_redacted'
-    data_dir = settings.CAP_DATA_DIR_VAR
-    if whitelisted:
-        data_dir = settings.WHITELISTED_DATA_DIR
+def get_matching_case_xml(case_id):
+    try:
+        xml = CaseXML.objects.get(case_id=case_id)
+        return xml.orig_xml
+    except CaseXML.DoesNotExist:
+        logger.error("Case id mismatch", case_id)
 
-    return "{}/{}/casemets/{}_CASEMETS_{}{}".format(data_dir,
-                                                    cdirname,
-                                                    cdirname,
-                                                    cpgnumber,
-                                                    settings.API_CASE_FILE_TYPE)
 
+def create_zip(case_list):
+    # tmp file backed by RAM up to 10MB, then stored to disk
+    tmp_file = tempfile.SpooledTemporaryFile(10 * 2 ** 20)
+    with zipfile.ZipFile(tmp_file, 'w', zipfile.ZIP_DEFLATED) as archive:
+        for case in case_list:
+            archive.writestr(case.slug + '.xml', get_matching_case_xml(case.case_id))
 
-def gzip_documents(zipname, filenames):
-    import zipfile
-    try:
-        compression = zipfile.zip_deflated
-    except:
-        compression = zipfile.ZIP_STORED
+    # Reset file pointer
+    tmp_file.seek(0)
 
-    with zipfile.ZipFile(zipname, 'w') as zapfile:
-        for f in filenames:
-            fname = f.split('/')[-1]
-            zapfile.write(f, fname, compress_type=compression)
+    # return open file handle
+    return tmp_file
 
-        return zapfile
 
+def create_zip_filename(case_list):
+    ts = slugify(datetime.now().timestamp())
+    if len(case_list) == 1:
+        return case_list[0].slug + '-' + ts + '.zip'
 
-def move_casezip(filename):
-    new_dest = "%s/%s" % (settings.CASE_ZIP_DIR, filename)
-    os.rename(filename, new_dest)
+    return '{0}_{1}_{2}.zip'.format(case_list[0].slug[:20], case_list[len(case_list)-1].slug[:20], ts)
```

Comment:
Nit: `case_list[len(case_list)-1]` is simpler as just `case_list[-1]`

Reply:
oh my. Brainfart.